auth/src/main/java/org/aerogear/auth/impl/OIDCAuthenticatorImpl.java

```diff
@@ -0,0 +1,26 @@
+package org.aerogear.auth.impl;
+
+import org.aerogear.auth.AuthServiceConfig;
+import org.aerogear.auth.credentials.ICredential;
+import org.aerogear.auth.credentials.OIDCCredentials;
+
+import java.security.Principal;
+
+/**
+ * Authenticates the user by using OpenID Connect.
+ */
+public class OIDCAuthenticatorImpl extends OIDCTokenAuthenticatorImpl {
```

Comment:
For someone with existing knowledge of OIDC, this might be a bit too vague as to what OAuth flow this is referring to. Might be better to have `AuthorisationCode` mentioned somewhere in the class name, as this is the name of the flow we are using here. Maybe `OIDCAuthCodeImpl`might work.

Reply:
good point!